Automattic/sensei
# Add patterns filter by post type

Fixes #5187

### Changes proposed in this Pull Request

* After a deeper investigation of the mentioned issue, it seems safe to use the `current_screen` hook to register the patterns based on the current post type. It wasn't really working on some beta versions, but it was fixed after the final release with [this commit](https://github.com/WordPress/WordPress/commit/571e1e01b3f7c11c8f8d6717be8568facd7ba118) (basically, the server adds our registered patterns to the initial setup of the patterns - writing it in the HTML, and the other come later in the async fetch). Anyway, I asked for a confirmation [here](https://github.com/WordPress/gutenberg/issues/40305#issuecomment-1139962066) just to make sure I'm not missing anything.

### Testing instructions


* In WP 6.0, create a new Course, and make sure only 3 patterns are listed in the patterns.
* Create a new Lesson, and make sure the other 3 patterns are listed in the patterns.
* It's nice to also test in WP 5.8 and 5.9.
